posydon/grids/psygrid.py

```diff
@@ -1544,6 +1544,8 @@ def __getitem__(self, idx):
         """Return a PSyRunView instance for the run with index `idx`."""
         if idx not in self:
             raise KeyError("Index {} out of bounds.".format(idx))
+        if not np.issubdtype(type(idx), int):
+            raise TypeError("Index {} is not of type int".format(idx))
```

Comment:
Should we move this to the `__contains__` function or add a similar statement there (in the second case, this needs to be moved two lines up)?
It feels wrong to me to get `a in grid == True`, but `grid[a]` raises an error.

Reply:
I see.
To verify: do you mean that the comparison can be made with a single-entry numpy array for `a in grid`, while we now force that to not work for `__getitem__`?

The same behaviour is preferable.

Reply:
Yes, I think we should have the same restrictions to both functions.

Reply:
I suggest utilizing `index` rather than `idx` in this function to maintain familiarity between similar functionalities in the code (like `__contains__` which currently uses the former).